src/main/kotlin/org/rust/ide/refactoring/changeSignature/RsChangeSignatureUsageProcessor.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.changeSignature
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Ref
+import com.intellij.psi.PsiElement
+import com.intellij.refactoring.changeSignature.ChangeInfo
+import com.intellij.refactoring.changeSignature.ChangeSignatureUsageProcessor
+import com.intellij.refactoring.rename.ResolveSnapshotProvider
+import com.intellij.usageView.UsageInfo
+import com.intellij.util.containers.MultiMap
+import org.jetbrains.annotations.NotNull
+import org.rust.RsBundle
+import org.rust.ide.presentation.getPresentation
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.ext.*
+
+class RsChangeSignatureUsageProcessor : ChangeSignatureUsageProcessor {
+    override fun findUsages(changeInfo: ChangeInfo?): Array<UsageInfo> {
+        val rsChangeInfo = changeInfo as? RsSignatureChangeInfo ?: return emptyArray()
+        val function = rsChangeInfo.config.function
+        val usages = findFunctionUsages(function).toMutableList()
+        if (function.owner is RsAbstractableOwner.Trait) {
+            function.searchForImplementations().filterIsInstance<RsFunction>().forEach { method ->
+                usages.add(RsFunctionUsage.MethodImplementation(method))
+                usages.addAll(findFunctionUsages(method))
+            }
+        }
+
+        return usages.toTypedArray()
+    }
+
+    override fun findConflicts(changeInfo: ChangeInfo?, refUsages: Ref<Array<UsageInfo>>): MultiMap<PsiElement, String> {
+        val rsChangeInfo = changeInfo as? RsSignatureChangeInfo ?: return MultiMap.empty()
+        val map = MultiMap.create<PsiElement, String>()
+        val config = rsChangeInfo.config
+        val function = config.function
+
+        findNameConflicts(function, config, map)
+        findVisibilityConflicts(function, config, refUsages.get(), map)
+
+        return map
+    }
+
+    override fun processUsage(
+        changeInfo: ChangeInfo?,
+        usageInfo: UsageInfo?,
+        beforeMethodChange: Boolean,
+        usages: Array<out UsageInfo>?
+    ): Boolean {
+        if (beforeMethodChange) return false
+
+        val rsChangeInfo = changeInfo as? RsSignatureChangeInfo ?: return false
+        val usage = usageInfo as? RsFunctionUsage ?: return false
+        val config = rsChangeInfo.config
+        if (usage is RsFunctionUsage.MethodImplementation) {
+            processFunction(config.function.project, config, usage.overriddenMethod)
+        }
+        else {
+            processFunctionUsage(config, usage)
+        }
+
+        return false
```

Comment:
```suggestion
        return true
```
`ChangeSignatureProcessorBase` expects that `true` is returned on success (though it will not change anything because there is just single `ChangeSignatureUsageProcessor`)

Reply:
Oh, ok. I thought that I should let it process other processors if there would be more in the future.

Reply:
Any other languages have single processor, so looks like it is unlikely that we will have more in future. Also please return `true` from `processPrimaryMethod` for same reason